ES|QL change_point docs and tech preview

jan-elastic opened the pull request.

jan-elastic committed:
ES|QL change point docs

jan-elastic committed:
Move ES|QL change_point to tech preview

craigtaverner commented on docs/reference/query-languages/esql/esql-commands.md:
```diff
@@ -31,6 +31,7 @@ An {{esql}} source command produces a table, typically with data from {{es}}. An
 
 {{esql}} supports these processing commands:
 
+* [`CHANGE_POINT`](#esql-change-point)
```
Should this have the `[preview]` prefix? I see you have the preview warning below, so I think perhaps the preview prefix should be here too, like it is for LOOKUP JOIN and MV_EXPAND.

craigtaverner reviewed (approved):
Looks fine, but I think needs the `[preview]` prefix. I'm restructuring these docs in another PR, but can deal with this change if you merge first.

craigtaverner commented:
We're also moving to make sure all examples in the commands docs are actually tested examples from the CSV-spec files. Perhaps you could add a comment about which file and which example is used in these docs?

craigtaverner commented:
I did a search for `| CHANGE_POINT message_count ON day` in the change_point.csv-spec file but could not find this example. Perhaps you could add the example to that file, or use an existing example from that file?

jan-elastic committed:
Update docs/reference/query-languages/esql/esql-commands.md

Co-authored-by: Craig Taverner <craig@amanzi.com>

craigtaverner commented:
We would like all examples in the docs to be examples from real tests, so that if, for example, a feature changes behaviour, the example changes too. In 8.x branches all docs examples are extracted from csv-spec files. In 9.x docs I've recovered that status for all the functions and operators, but not yet for the commands. So I'm working on making commands use csv-spec examples, and all those that already existed in 8.x will be fine, but any new ones (like yours) might not be fine.

jan-elastic committed:
different example + add it the csv tests

jan-elastic commented:
OK, I have added a different example and also added that one to the csv tests.

I didn't feel like adding MBs of data to the test suite, even though I think the `kibana_sample_data_logs` gives a nicer example for the documentation.

jan-elastic committed:
Merge branch 'main' into esql_change_point_docs_and_tech_preview

luigidellaquila commented on x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java:
```diff
@@ -843,7 +843,7 @@ public enum Cap {
          * Support change point detection "CHANGE_POINT".
          */
-        CHANGE_POINT(Build.current().isSnapshot()),
+        CHANGE_POINT,
```
If the intention is to have it in non-snapshot releases, I think we also have to change the [grammar](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4#L61)

craigtaverner commented on docs/reference/query-languages/esql/esql-commands.md:
```diff
@@ -191,6 +192,69 @@ SHOW INFO
 | 8.13.0 | 2024-02-23T10:04:18.123117961Z | 04ba8c8db2507501c88f215e475de7b0798cb3b3 |
 
 
+## `CHANGE_POINT` [esql-change-point]
```
Since the merger of https://github.com/elastic/elasticsearch/pull/126279, this now needs to be done differently. You need to move this entire section into a file called `_snippets/commands/layout/change_point.md` and include that from the file `commands/processing_commands.md`. Then you need to edit the change_point.md file to include the right tests from `_snippets/commands/examples/*`.

craigtaverner reviewed (changes requested):
This needs to be modified a lot for the new split-up commands docs.

craigtaverner committed:
Merge remote-tracking branch 'origin/main' into esql_change_point_docs_and_tech_preview

craigtaverner commented on docs/reference/query-languages/esql/esql-commands.md:
```diff
@@ -31,6 +31,7 @@ An {{esql}} source command produces a table, typically with data from {{es}}. An
 
 {{esql}} supports these processing commands:
 
+* [preview] [`CHANGE_POINT`](#esql-change-point)
```
To match other commands with `_` characters, this link should be `esql-change_point` (see `esql-mv_expand` for example).

jan-elastic committed:
fix grammar

jan-elastic committed:
License check

craigtaverner committed:
Restructure change_point docs to new structure

craigtaverner committed:
Added generated test examples to change_point docs

craigtaverner committed:
Fixed a few README.md text mistakes and added more details

craigtaverner reviewed (approved):
Looks good to me.

jan-elastic committed:
regen parser

elasticsearchmachine commented:
Pinging @elastic/kibana-esql (ES|QL-ui)

stratoula commented:
@darnautov when this is merged we can enable the change_pont autocomplete

craigtaverner commented on docs/reference/query-languages/esql/_snippets/commands/layout/change_point.md:
```diff
@@ -0,0 +1,58 @@
+## `CHANGE_POINT` [esql-change_point]
+
+:::{note}
+The `CHANGE_POINT` function requires a [platinum license](https://www.elastic.co/subscriptions).
```
```suggestion
The `CHANGE_POINT` command requires a [platinum license](https://www.elastic.co/subscriptions).
```

craigtaverner reviewed (approved):
Looks good to me. One change suggested in the docs where we should call `CHANGE_POINT` a command, not a function.

jan-elastic committed:
Update docs/reference/query-languages/esql/_snippets/commands/layout/change_point.md

Co-authored-by: Craig Taverner <craig@amanzi.com>

jan-elastic committed:
Merge branch 'main' into esql_change_point_docs_and_tech_preview

jan-elastic committed:
Merge branch 'main' into esql_change_point_docs_and_tech_preview
